src/Microsoft.AspNetCore.Server.IISIntegration/Server/HttpProtocol.cs

```diff
@@ -193,14 +208,25 @@ private IISAwaitable DoFlushAsync()
             unsafe
             {
                 var hr = 0;
-                hr = NativeMethods.http_flush_response_bytes(_pHttpContext, IISAwaitable.FlushCallback, (IntPtr)_thisHandle, out var fCompletionExpected);
-
-                if (!fCompletionExpected)
+                if (_wasUpgraded)
                 {
-                    CompleteFlush(hr, 0);
+                    hr = NativeMethods.http_websockets_flush_bytes(_pHttpContext, IISAwaitable.FlushCallback, (IntPtr)_thisHandle, out var fCompletionExpected);
```

Comment:
When does this call happen? Once as part of upgrade?

Reply:
It shouldn't be called at all. Will remove.